tools/MultiProjectValidator/AnalysisRules/DependencyMismatchRule.cs

```diff
@@ -3,24 +3,30 @@
 using System.Linq;
 using Microsoft.DotNet.ProjectModel;
 using System.Text;
-using ProjectSanity.AnalysisRules.DependencyMismatch;
+using MultiProjectValidator.AnalysisRules.DependencyMismatch;
 
 namespace MultiProjectValidator.AnalysisRules
 {
     public class DependencyMismatchRule : IAnalysisRule
     {
-        public AnalysisResult Evaluate(List<ProjectContext> projectContexts)
+        public AnalysisResult Evaluate(IEnumerable<ProjectContext> projectContexts)
         {
-            var targetGroupedContexts = GroupContextsByTarget(projectContexts);
+            var filteredContexts = FilterContextList(projectContexts);
+            var targetGroupedContexts = GroupContextsByTarget(filteredContexts);
 
             var failureMessages = EvaluateProjectContextTargetGroups(targetGroupedContexts);
-            var pass = failureMessages.Count == 0;
+            var pass = failureMessages.Count() == 0;
 
             var result = new AnalysisResult(failureMessages, pass);
             return result;
         }
 
-        private List<string> EvaluateProjectContextTargetGroups(Dictionary<string, List<ProjectContext>> targetGroupedProjectContexts)
+        private IEnumerable<ProjectContext> FilterContextList(IEnumerable<ProjectContext> projectContexts)
+        {
+            return projectContexts.Where(context=> !context.TargetIsDesktop()).Select(context => context);
```

Comment:
My mental model of LINQ is mostly based of SQL :smile: 

Will make the change.